src/form/form-input.tsx

```diff
@@ -45,8 +50,8 @@ const Component = (props: FormInputProps, ref: InputRef): JSX.Element => {
             xxlCols={xxlCols}
         >
             <Input
-                id={`${id}-base`}
-                data-testid={testId || id}
+                id={`${inputId}-base`}
+                data-testid={testId || inputId}
```

Comment:
and the input will receive the suffixed id and data-testid

```suggestion
                data-testid={testId ? `${testId}-base` : undefined }
```